packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -24,15 +28,66 @@ import {
 	type VisualTreeNode,
 } from "./VisualTree.js";
 
+/**
+ * Returns allowed types of the non-leaf nodes in the tree.
+ * - String: Allowed type for the array node.
+ * - `Record<string, string>`: Allowed types for the non-array nodes (e.g., object, map).
+ */
+function createAllowedTypesVisualTree(
+	allowedTypes: string | Record<string, string>,
+): VisualChildNode {
+	if (typeof allowedTypes === "string") {
+		return {
+			value: allowedTypes,
+			nodeKind: VisualNodeKind.ValueNode,
+		};
+	}
+
+	const result: Record<string, VisualValueNode> = {};
+	for (const [allowedTypeKey, allowedType] of Object.entries(allowedTypes)) {
+		result[allowedTypeKey] = {
+			nodeKind: VisualNodeKind.ValueNode,
+			value: allowedType,
+		};
+	}
+
+	return {
+		children: result,
+		nodeKind: VisualNodeKind.TreeNode,
+	};
+}
+
+/**
+ * Creates a visual representation of the schema of the tree in {@link VisualTreeNode} format.
+ */
+function createToolTipContents(tree: SharedTreeSchemaNode): VisualTreeNode {
+	const children: Record<string, VisualChildNode> = {
+		name: {
+			nodeKind: VisualNodeKind.ValueNode,
+			value: tree.schemaName,
+		},
+	};
+	if (tree.allowedTypes !== undefined) {
+		children.allowedTypes = createAllowedTypesVisualTree(tree.allowedTypes);
+	}
+	return {
+		nodeKind: VisualNodeKind.TreeNode,
+		children,
+	};
+}
+
 /**
  * Converts the output of {@link sharedTreeVisualizer} to {@link VisualChildNode} type containing `schema` and `children` fields.
+ * Constructs a VisualTree of the input tree's schema fields in {@link VisualTreeNode} or {@link VisualValueNode}.
```

Comment:
I think we want to get rid of the first sentence here? These seem sort of redundant, but the second sentence seems more accurate.